craft_providers/bases/buildd.py

```diff
@@ -78,6 +78,57 @@ def _check_deadline(
         raise BaseConfigurationError(brief=message)
 
 
+def _network_connected(executor):
+    """Check if the network is connected."""
+    # check if the port is open using bash's built-in tcp-client (1.1.1.1 is a well
+    # known DNS server, 53 is the DNS port)
+    command = ["bash", "-c", "exec 3<> /dev/tcp/1.1.1.1/53"]
```

Comment:
1.1.1.1 will be blocked in many enterprises which is why we went with getent https://github.com/canonical/craft-providers/pull/176/files#diff-bcc16e27108e1d381f59fe92f1439c56ac920252ab7be057491af4a20bcb7748R817

Reply:
You're right about `1.1.1.1` maybe blocked. 

I don't like the `getent` solution because it's too indirect, some systems may not have that mechanism, and mainly because we're currently using it to assert that the network is already there before continuing with the setup, which then fails because of missing network.

After a conversation with Igor and Callahan we decided that we could improve this verification by actually referring our site, using the HTTPS port.